src/Sulu/Bundle/MediaBundle/DependencyInjection/SuluMediaExtension.php

```diff
@@ -407,32 +409,27 @@ public function load(array $configs, ContainerBuilder $container)
             ]
         );
 
-        $this->configureStorage($config, $container, $loader);
         $this->configureFileValidator($config, $container);
 
         $container->registerForAutoconfiguration(MediaPropertiesProviderInterface::class)
             ->addTag('sulu_media.media_properties_provider');
     }
 
-    private function configureStorage(array $config, ContainerBuilder $container, LoaderInterface $loader)
+    private function configureStorage(array $config, ContainerBuilder $container): void
     {
-        $storage = $container->resolveEnvPlaceholders($config['storage'], true);
-        $container->setParameter('sulu_media.media.storage', $storage);
-
-        foreach ($config['storages'] as $storageKey => $storageConfig) {
-            foreach ($storageConfig as $key => $value) {
-                if ($storageKey === $storage) {
-                    $container->setParameter('sulu_media.media.storage.' . $storageKey . '.' . $key, $value);
-                } else {
-                    // Resolve unused ENV Variables of other Adapter
-                    $container->resolveEnvPlaceholders($value, true);
-                }
-            }
-        }
-
-        $loader->load('services_storage_' . $storage . '.xml');
+        $sorageServiceId = $config['storage']['flysystem_service'];
+        $adapterService = 'flysystem.adapter.' . $sorageServiceId;
+
+        $container->register('sulu_media.storage', FlysystemStorage::class)
+            ->setArguments([
+                new Reference($sorageServiceId),
+                new Reference($adapterService),
+                $config['storage']['segments'],
+                null,
```

Comment:
add comment:
```suggestion
                null, // defined via compilerpass @see FlysystemCompilerPass
```

add also the `use ...\FlysystemCompilerPass` it can be clicked on the class name

Reply:
Comment added, but we don't need a use statement because the class is in the same namespace.